### PR TITLE
Fix GetSnapshots to not return non-existent snapshots with ignore_unavailable=true

### LINKED ISSUE #6820: [BUG] Get snapshot API returns incorrect response
**Describe the bug**
When calling Get snapshot API with setting `ignore_unavailable` parameter to `true`, if the snapshot does not exist, then this API will return all of the executing snapshots which are in `IN_PROGRESS` state.

**To Reproduce**
1. Create a snapshot, which will take a long time to execute.
2. Call the Get Snapshot API with an non-existing snapshot and setting `ignore_unavailable` parameter to `true`:
`GET _snapshot/test/xx?ignore_unavailable=true`
then you can see that the response contains the snapshot which is executing and in `IN_PROGRESS` state.

**Expected behavior**
Return empty response if the snapshot does not exist.

**Screenshots**
image

**Host/Environment (please complete the following information):**
 - OS: [Centos]
 - Version [2.4.1]

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
Relates to #6820.

Currently, the Get Snapshot API returns all of the snapshots in-progress when the specified snapshot is non-existent and the parameter `ignore_unavailable` is  set to `true`.

The main changes are:
1. Fix the bug for Get Snapshot API, return empty result when getting a non-existing snapshot with setting `ignore_unavailable` to `true` and there are just some snapshots in-progress. 
2. Add some test code for the change.

### Issues Resolved
#6820.

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Shouldn't specifying a non-existent snapshot be an error?

**Comment by gaobinlong** [comment]
@dblock just like `Get index API` or `Get alias API`, I think the common parameter `ignore_unavailable` contains the meaning of `ignoring missing`, by going through the code of `Get Snapshot API` I found that's the same logic, when the specified snapshot is non-existent, the api will throw `SnapshotMissingException` if `ignore_unavailable` is `false`(by default), but return empty result if `ignore_unavailable` is `true` (has a bug when there are some snapshots in-progress, this is what this PR wants to fix).

File: CHANGELOG.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Added equals/hashcode for named DocValueFormat.DateTime inner class ([#6357](https://github.com/opensearch-project/OpenSearch/pull/6357))
 - Fixed bug for searchable snapshot to take 'base_path' of blob into account ([#6558](https://github.com/opensearch-project/OpenSearch/pull/6558))
 - Fix fuzziness validation ([#5805](https://github.com/opensearch-project/OpenSearch/pull/5805))
+- Fix bug for Get Snapshot API to return correct response when getting a non-existing snapshot ([#6839](https://github.com/opensearch-project/OpenSearch/pull/6839))
```
Comment (dblock) [inline CHANGELOG.md:115]: Let's get more specific: "Fix `GetSnapshots` returning non-existent snapshots"?

**Review by dblock — APPROVED** [review]
Thanks for the clarification. Is there a unit test possible for this?

File: CHANGELOG.md (hunk 1 above)
Comment (gaobinlong) [reply CHANGELOG.md:115]: Sure, I will change it.

**Comment by gaobinlong** [comment]
> Thanks for the clarification. Is there a unit test possible for this?

Thanks for your review and suggestion, I've tried adding unit tests for the changed method but found `repositoryService` is referenced in the method and is hard to mock: https://github.com/opensearch-project/OpenSearch/blob/53b128fc10f4b2ee5c95b55000dbfa89f110a661/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L247, so maybe we can only add integration test for this change and it seems that most snapshots related operations only have integration test.

**Review by dblock — APPROVED** [review]
Let's merge. Consider some refactoring to make unit tests possible.

**Pull request merged**